python/experiments.py

```diff
@@ -29,6 +30,11 @@ def task_compile_benchmarks():
     for benchmark in manifest["benchmarks"]:
         filepath = util.churchroad_evaluation_dir() / benchmark["filepath"]
         benchmark_name = filepath.stem
+        _benchmark_flags = benchmark.get("flags") if benchmark["flags"] else ""
+        _features = benchmark.get("features") if benchmark["features"] else {}
+        _extra_summary_fields = {"tool": "vivado", "name": benchmark_name}
+        for feature in _features:
+            _extra_summary_fields[feature] = _features.get(feature)
```

Comment:
1. You don't need underscores before these variable names. In general, underscores are used (1) before "private" functions (i.e. functions that shouldn't be used outside of the file) or private class member variables and (2) before unused variables. These are neither!
2. Your if statements here are off. I think you probably want e.g. `benchmark["flags"] if "flags" in benchmark else ""`

Reply:
I agree with (1)!

For (2), isn't `benchmark["flags"]` and `benchmark.get("flags")` equivalent? I assumed they would be `True` if `benchmark["flags"]` is non-empty.